### PR TITLE
XIVY-10312 Use business states of task and cases

### PR DESCRIPTION
**Pull request by jackra1n**
image

### CODE REVIEW

File: dev-workflow-ui/webContent/includes/components/casesTable.xhtml [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
     <p:column filterBy="#{case.state}" styleClass="icon-row" sortBy="#{case.state}">
       <h:outputText id="caseState" styleClass="si si-lg si-#{casesIvyDevWfBean.getStateIcon(case.state)}" />
-      <p:tooltip for="caseState" value="State: #{case.state}" position="top" />
+      <p:tooltip for="caseState" value="State: #{case.businessState} (#{case.state})" position="top" />
```
Comment (alexsuter) [inline dev-workflow-ui/webContent/includes/components/casesTable.xhtml:25]: use here case.state and case.technicalState as well

File: dev-workflow-ui/webContent/includes/components/taskDetails/description.xhtml [hunk 2]
```diff
@@ -20,10 +20,19 @@
...
     </h:outputText>
 
+    <h:outputText value="Category" />
+    <h:outputText value="#{tasksDetailsIvyDevWfBean.selectedTask.category}" id="category" />
+
     <h:outputText value="State" />
     <h:outputText>
       <i class="si si-#{tasksDetailsIvyDevWfBean.selectedTask.stateIcon} p-mr-1" />
-      <h:outputText value="#{tasksDetailsIvyDevWfBean.selectedTask.state}" id="taskState" />
+      <h:outputText value="#{tasksDetailsIvyDevWfBean.selectedTask.technicalState}" id="taskState" />
```
Comment (alexsuter) [inline dev-workflow-ui/webContent/includes/components/taskDetails/description.xhtml:29]: wrong.

File: dev-workflow-ui/webContent/includes/components/taskDetails/description.xhtml [hunk 3]
```diff
@@ -20,10 +20,19 @@
...
+    </h:outputText>
+
+    <h:outputText value="Technical state" />
+    <h:outputText>
+      <i class="si si-#{tasksDetailsIvyDevWfBean.selectedTask.stateIcon} p-mr-1"></i>
+      <h:outputText value="#{tasksDetailsIvyDevWfBean.selectedTask.state}" id="taskTechnicalState" />
```
Comment (alexsuter) [inline dev-workflow-ui/webContent/includes/components/taskDetails/description.xhtml:35]: wrong

File: dev-workflow-ui/webContent/resources/components/caseLink.xhtml [hunk 4]
```diff
@@ -11,7 +11,7 @@
     <f:param name="case" value="#{cc.attrs.case.id}"/>
     <i class="si si-task-list-approve p-mr-1" />
     <h:outputText id="caseState" styleClass="si si-#{casesIvyDevWfBean.getStateIcon(cc.attrs.case.getState())} p-mr-1"/>
-    <p:tooltip for="caseState" value="State: #{cc.attrs.case.state}" position="top"/>
+    <p:tooltip for="caseState" value="State: #{cc.attrs.case.businessState} (#{cc.attrs.case.state})" position="top"/>
```
Comment (alexsuter) [inline dev-workflow-ui/webContent/resources/components/caseLink.xhtml:14]: case.state (case.technicalState)

**Review by alexsuter — CHANGES_REQUESTED** [review]

File: dev-workflow-ui/webContent/includes/components/casesTable.xhtml (hunk 1 above)
Comment (jackra1n) [reply dev-workflow-ui/webContent/includes/components/casesTable.xhtml:25]: @alexsuter the problem is that this is `ICase` so it has only `businessState` and `state`
I could change the `TaskModel` to be the same as this tho?

Comment (alexsuter) [reply dev-workflow-ui/webContent/includes/components/casesTable.xhtml:25]: CaseModel could be anyway a good idea. But yes please change TaskModel, that this is the same.

**Review by alexsuter — APPROVED** [review]

**Pull request merged**